### PR TITLE
Create additional fields for hbz publishing catalogue

### PR DESCRIPTION
**Pull request by TobiasNx**
Related to #6. Create tags for specific hbz catalogue.

I am not sure:
- How to reuse the GND Elements without copying them.
-  - Tried it like this: https://github.com/hbz/qa-catalogue/pull/1/commits/517869aecd95913a937935427edb4c305c4bef24

### CODE REVIEW

**Comment by TobiasNx** [comment]
I am not sure how we can reuse the DNB tags without copying them?

**Comment by Phu2** [comment]
I'm not sure either if https://github.com/hbz/qa-catalogue/pull/1/commits/517869aecd95913a937935427edb4c305c4bef24#diff-6ab233441d2457fd4d4370036d454637fc9ac8133b4d0226ffffa3b206391d18 will work...

We should resolve the fest failures first, then try it out. What do you think?

**Comment by Phu2** [comment]
Disabled Qodana workflow as set in the upstream repo. See manual https://docs.github.com/de/actions/managing-workflow-runs-and-deployments/managing-workflow-runs/disabling-and-enabling-a-workflow

**Comment by TobiasNx** [comment]
> I'm not sure either if [517869a#diff-6ab233441d2457fd4d4370036d454637fc9ac8133b4d0226ffffa3b206391d18](https://github.com/hbz/qa-catalogue/commit/517869aecd95913a937935427edb4c305c4bef24#diff-6ab233441d2457fd4d4370036d454637fc9ac8133b4d0226ffffa3b206391d18) will work...
> 
> We should resolve the fest failures first, then try it out. What do you think?

Sounds good!

**Comment by Phu2** [comment]
I think we still have to follow this step "c. modify the expected number of data elements at src/test/java/de/gwdg/metadataqa/marc/utils/DataElementsStaticticsTest.java" in the [README](https://github.com/hbz/qa-catalogue?tab=readme-ov-file#appendix-ii-handling-marc-versions) but don't know how to identify this number. @TobiasNx Any clue?

**Comment by TobiasNx** [comment]
https://github.com/hbz/qa-catalogue/actions/runs/10469385366/job/28992298463?pr=1#step:4:16664

File: src/test/java/de/gwdg/metadataqa/marc/utils/MarcTagListerTest.java [hunk 1]
```diff
@@ -28,7 +28,7 @@ public void testListTag() {
     List<Class<? extends DataFieldDefinition>> tags = MarcTagLister.listTags();
     assertNotNull(tags);
     assertNotEquals(0, tags.size());
-    assertEquals(457, tags.size());
+    assertEquals(493, tags.size());
```
[the remark this answers is not in the document]
Comment (TobiasNx) [reply src/test/java/de/gwdg/metadataqa/marc/utils/MarcTagListerTest.java:31]: logs in the build.

**Comment by Phu2** [comment]
Next test failure:

`Failed tests:   testPatternMatchingWithSubfield(de.gwdg.metadataqa.marc.analysis.bl.UseCaseTest): expected:<[110$e, 111$e, 100$e, 700$e, 710$e, 711$e, 720$e, 751$e, 752$e, 775$e, 788$e]> but was:<[100$e, 110$e, 111$e, 700$e, 710$e, 711$e, 720$e, 751$e, 752$e, 775$e, 788$e]>`

https://github.com/hbz/qa-catalogue/actions/runs/10472986275/job/29003791025#step:4:16625

**Comment by TobiasNx** [comment]
> Next test failure:
> 
> `Failed tests: testPatternMatchingWithSubfield(de.gwdg.metadataqa.marc.analysis.bl.UseCaseTest): expected:<[110$e, 111$e, 100$e, 700$e, 710$e, 711$e, 720$e, 751$e, 752$e, 775$e, 788$e]> but was:<[100$e, 110$e, 111$e, 700$e, 710$e, 711$e, 720$e, 751$e, 752$e, 775$e, 788$e]>`
> 
> [hbz/qa-catalogue/actions/runs/10472986275/job/29003791025#step:4:16625](https://github.com/hbz/qa-catalogue/actions/runs/10472986275/job/29003791025#step:4:16625)
> 
> @TobiasNx `100$e` is expected at 3rd position, but it's at 1st ??

I am confused, I think I did not change anything that touches this test. But I will have another look.

File: src/test/java/de/gwdg/metadataqa/marc/utils/MarcTagListerTest.java [hunk 2]
```diff
@@ -117,6 +117,9 @@ public void testListTag() {
     assertEquals( 12, (int) versionCounter2.get(MarcVersion.OGYK));
     assertEquals( 12, (int) versionCounter.get("ogyktags"));
 
+    assertEquals( 36, (int) versionCounter2.get(MarcVersion.HBZ));
```
Comment (Phu2) [inline src/test/java/de/gwdg/metadataqa/marc/utils/MarcTagListerTest.java:120]: @TobiasNx Same with this number? (I don't even understand what it is about...)

File: src/test/java/de/gwdg/metadataqa/marc/utils/MarcTagListerTest.java [hunk 3]
```diff
@@ -117,6 +117,9 @@ public void testListTag() {
...
+    assertEquals( 36, (int) versionCounter.get("hbztags"));
```
Comment (Phu2) [inline src/test/java/de/gwdg/metadataqa/marc/utils/MarcTagListerTest.java:121]: @TobiasNx Same with these numbers? (I don't even understand what they are about...)

Comment (TobiasNx) [reply src/test/java/de/gwdg/metadataqa/marc/utils/MarcTagListerTest.java:121]: Counted my new files in the hbztags folder :D

**Comment by TobiasNx** [comment]
I applied the changes suggested by @pkiraly in https://github.com/pkiraly/qa-catalogue/issues/503#issuecomment-2301686469

**Comment by pkiraly** [comment]
Now the build is failed because the workflow has steps to send code to external tools, including Sonar, a code quality analysis tool. It requires a token (SONAR_TOKEN) defined in the repository. To be honest I do not know what is the best practice on that: sharing this token, letting others to create token for themselves, or disabling this step in the Github action. Maybe I should move this into another workflow that you can disable. Anyway: there are no test failures, which is great.

**Comment by Phu2** [comment]
Thank you for your help, @pkiraly !

**Comment by TobiasNx** [comment]
These elements should checked too.
- [ ] 009	https://service-wiki.hbz-nrw.de/pages/viewpage.action?pageId=525369446
- [ ] 009	https://service-wiki.hbz-nrw.de/pages/viewpage.action?pageId=525369446
- [ ] 029	https://service-wiki.hbz-nrw.de/pages/viewpage.action?pageId=525369446
- [ ] 089	https://service-wiki.hbz-nrw.de/pages/viewpage.action?pageId=525369446
- [ ] 249(?) https://service-wiki.hbz-nrw.de/pages/viewpage.action?pageId=525369446 checken ob noch abweicht
- [ ] 259(?) https://service-wiki.hbz-nrw.de/pages/viewpage.action?pageId=525369446 checken ob noch abweicht
- [ ] 290 https://service-wiki.hbz-nrw.de/pages/viewpage.action?pageId=525369446
- [ ] 589	https://service-wiki.hbz-nrw.de/pages/viewpage.action?pageId=525369446 checken ob noch abweicht
- [ ] 591	https://service-wiki.hbz-nrw.de/pages/viewpage.action?pageId=525369446 checken ob noch abweicht
- [ ] 689(?) https://service-wiki.hbz-nrw.de/pages/viewpage.action?pageId=525369446 checken ob noch abweicht
- [ ] 889(?) https://service-wiki.hbz-nrw.de/pages/viewpage.action?pageId=525369446 checken ob noch abweicht
- [ ] 912 https://service-wiki.hbz-nrw.de/pages/viewpage.action?pageId=525369446 
- [ ] 924(?)  https://service-wiki.hbz-nrw.de/pages/viewpage.action?pageId=525369446 
- [ ] 960	 https://service-wiki.hbz-nrw.de/pages/viewpage.action?pageId=525369446 
- [ ] 961  https://service-wiki.hbz-nrw.de/pages/viewpage.action?pageId=525369446 
- [ ] 962	 https://service-wiki.hbz-nrw.de/pages/viewpage.action?pageId=525369446 
- [ ] 964	 https://service-wiki.hbz-nrw.de/pages/viewpage.action?pageId=525369446 
- [ ] 965	 https://service-wiki.hbz-nrw.de/pages/viewpage.action?pageId=525369446 
- [ ] 980 https://service-wiki.hbz-nrw.de/pages/viewpage.action?pageId=518750316
- [ ] 981 https://service-wiki.hbz-nrw.de/pages/viewpage.action?pageId=518750316
- [ ] 982 https://service-wiki.hbz-nrw.de/pages/viewpage.action?pageId=518750316
- [ ] 983 https://service-wiki.hbz-nrw.de/pages/viewpage.action?pageId=518750316
- [ ] 984 https://service-wiki.hbz-nrw.de/pages/viewpage.action?pageId=518750316
- [ ] 985 https://service-wiki.hbz-nrw.de/pages/viewpage.action?pageId=518750316
- [ ] 986 https://service-wiki.hbz-nrw.de/pages/viewpage.action?pageId=518750316
- [ ] 987 https://service-wiki.hbz-nrw.de/pages/viewpage.action?pageId=518750316
- [ ] 988 https://service-wiki.hbz-nrw.de/pages/viewpage.action?pageId=518750316
- [ ] 989 https://service-wiki.hbz-nrw.de/pages/viewpage.action?pageId=518750316
- [ ] 990 https://service-wiki.hbz-nrw.de/pages/viewpage.action?pageId=518750316
- [ ] 991 https://service-wiki.hbz-nrw.de/pages/viewpage.action?pageId=518750316
- [ ] 992 https://service-wiki.hbz-nrw.de/pages/viewpage.action?pageId=518750316
- [ ] 993 https://service-wiki.hbz-nrw.de/pages/viewpage.action?pageId=518750316
- [ ] 994 https://service-wiki.hbz-nrw.de/pages/viewpage.action?pageId=518750316
- [ ] 995 https://service-wiki.hbz-nrw.de/pages/viewpage.action?pageId=518750316
- [ ] 996 https://service-wiki.hbz-nrw.de/pages/viewpage.action?pageId=518750316
- [ ] 997 https://service-wiki.hbz-nrw.de/pages/viewpage.action?pageId=518750316
- [ ] 998 https://service-wiki.hbz-nrw.de/pages/viewpage.action?pageId=518750316
- [ ] 999 https://service-wiki.hbz-nrw.de/pages/viewpage.action?pageId=518750316

File: src/main/java/de/gwdg/metadataqa/marc/definition/tags/hbztags/TagH24.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
+package de.gwdg.metadataqa.marc.definition.tags.hbztags;
+
+import org.apache.hadoop.shaded.org.apache.commons.net.nntp.Article;
```
Comment (pkiraly) [inline src/main/java/de/gwdg/metadataqa/marc/definition/tags/hbztags/TagH24.java:3]: Are you sure something from Hadoop is needed here?

File: src/main/java/de/gwdg/metadataqa/marc/definition/tags/hbztags/TagH56.java [hunk 5]
```diff
@@ -0,0 +1,91 @@
+package de.gwdg.metadataqa.marc.definition.tags.hbztags;
+
+import org.apache.hadoop.shaded.org.jline.builtins.telnet.Telnet;
```
Comment (pkiraly) [inline src/main/java/de/gwdg/metadataqa/marc/definition/tags/hbztags/TagH56.java:3]: I guess it is a mistake.

File: src/main/java/de/gwdg/metadataqa/marc/definition/tags/hbztags/TagH24.java (hunk 4 above)
Comment (TobiasNx) [reply src/main/java/de/gwdg/metadataqa/marc/definition/tags/hbztags/TagH24.java:3]: I dont know how this got in there. Perhaps my editor did some unintended stuff.

**Comment by TobiasNx** [comment]
I added the DNB elements that are re-used in hbz again until https://github.com/pkiraly/qa-catalogue/issues/507 is solved.

**Comment by TobiasNx** [comment]
@Phu2 should we merge?

**Pull request merged**